nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-nar-utils/src/main/java/org/apache/nifi/nar/NativeLibFinder.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.nar;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+public interface NativeLibFinder {
+    Logger LOGGER = LoggerFactory.getLogger(NativeLibFinder.class);
+
+    String OS = System.getProperty("os.name").toLowerCase();
+
+    List<File> getNativeLibDirs();
+
+    Map<String, Path> getNativeLibNameToPath();
+
+    String getTmpLibFilePrefix();
+
+    default Set<File> getUsrLibDirs() {
```

Comment:
Rather than having a default function for this that is evaluated each time it's needed, these values, I don't believe, will change over the lifetime of the JVM. Would recommend instead making it into a `private static final Set<File>` and computing only once.

Reply:
Fair enough. Is a minor quibble. I would say that it's okay to keep it as-is, if you added just a single-line comment along the lines of `// Computed at runtime instead of statically to help facilitate testing. Not called often so the expense is worth the tradeoff` or something like that. Just to show that it has been thought about and a decision intentionally made....